ush/python/pygw/src/tests/test_schema.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+
+# ----
+
+"""
+Module
+------
+
+    test_schema.py
+
+Description
+-----------
+
+    This module provides a series of unit-tests for the pygw.schema
+    module.
+
+Usage
+-----
+
+    pytest tests/test_schema.py
+
+Functions
+---------
+
+    test_schema_float()
+
+        This function provides a unit test for the schema module; a
+        float-value will be passed to the schema validator that
+        accepts only string or integer instances; in order to pass,
+        this unit test must raise a a SchemaException.
+
+    test_schema_int()
+
+        This function provides a unit test for the schema module; a
+        integer-value is passed to the schema.
+
+    test_schema_opt()
+
+        This function provides a unit test for the schema module; a
+        integer-value and an optional string value `opt_var` are
+        passed to the schema.
+
+    test_schema_str()
+
+        This function provides a unit test for the schema module; a
+        string is passed to the schema.
+
+Requirements
+------------
+
+schema; https://github.com/keleshev/schema
+
+Author(s)
+---------
+
+    Henry R. Winterbottom; 20 March 2023
+
+History
+-------
+
+    2023-03-20: Henry Winterbottom -- Initial implementation.
+
+"""
+
+# ----
+
+from pygw.exceptions import SchemaException
+from pygw.logger import Logger, logit
+from pygw.schema import validate_opts
+from schema import Optional, Or
+
+# ----
+
+base_logger = Logger(level="info", colored_log=True)
+
+# ----
+
+# Define the schema.
+CLS_SCHEMA = {
+    "check_var": Or(str, int),
+    Optional("opt_var"): str,
+}
+
+# ----
+
+
+@logit(base_logger)
+def test_schema_float() -> None:
+    """
+    Description
+    -----------
+
+    This function provides a unit test for the schema module; a
+    float-value will be passed to the schema validator that accepts
+    only string or integer instances; in order to pass, this unit test
+    must raise a a SchemaException.
+
+    """
+
+    # Define the option values to be evaluated within the context of
+    # the specified schema.
+    cls_opts = {"check_var": 0.0000}
+
+    # Evaluate the schema.
+    try:
+        validate_opts(cls_schema=CLS_SCHEMA, cls_opts=cls_opts)
+
+    except SchemaException:
+        pass
+
```

Comment:
This will also incorrectly pass if no exception is raised.

Reply:
The `schema` library will also raise it's own exception (which it does for this unit-test). I am just capturing the exception in a `pygw` specific exception. I want the exception to be raised (and thus pass) since I am proving the schema incorrectly formatted attributes.

Reply:
I get that, that's not the problem. The issue is if `validate_opts` does NOT raise an exception, the `except` clause is passed over and the function will still return successfully. Since this unit test is confirming invalid output DOES raise an exception, the test should be failing if no exception is raised, not just if a different exception is raised. Recommend changing `pass` to `return` and then adding an `assert false` after the try/except block.

Reply:
Got it. Thank you for the explanation. That completely eluded me.